runelite-client/src/main/java/net/runelite/client/ui/overlay/components/MinimapOrb.java

```diff
@@ -0,0 +1,197 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay.components;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import java.awt.RadialGradientPaint;
+import java.awt.Shape;
+import java.awt.geom.Point2D;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import lombok.Setter;
+import net.runelite.api.Point;
+import net.runelite.client.ui.FontManager;
+
+public class MinimapOrb
+{
+	private static final int ORB_X = 26;
+	private static final int ORB_Y = 2;
+	private static final int ORB_W = 29;
+
+	private static final Color BACKGROUND_COLOR = new Color(20, 20, 20);
+	private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 50);
+
+	// black background
+	private final BufferedImage background;
+	// Color and reflection highlight
+	private final BufferedImage fill;
+	// Shadow for when enabled
+	private final BufferedImage fillEnabled;
+	// Highlight for recharge
+	private final BufferedImage highlight;
+	// Frame
+	private final BufferedImage frame;
+	// Icon to draw ontop
+	@Setter
+	private BufferedImage overlay;
+
+	public MinimapOrb(Color color, Color enabledColor, BufferedImage overlay)
+	{
+		{
+			background = new BufferedImage(ORB_W, ORB_W, BufferedImage.TYPE_INT_ARGB);
+			Graphics2D g = background.createGraphics();
+
+			g.setColor(BACKGROUND_COLOR);
+			g.fillOval(0, 0, ORB_W, ORB_W);
+		}
+
+		fill = createFill(color, false);
+		fillEnabled = createFill(enabledColor, true);
+
+		{
+			highlight = new BufferedImage(ORB_W, ORB_W, BufferedImage.TYPE_INT_ARGB);
+			Graphics2D g = highlight.createGraphics();
+
+			g.setColor(HIGHLIGHT_COLOR);
+			g.fillOval(0, 0, ORB_W, ORB_W);
+		}
+
+		try
+		{
+			frame = ImageIO.read(getClass().getResourceAsStream("minimap_orb_background.png"));
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+
+		this.overlay = overlay;
+	}
+
+	private BufferedImage createFill(Color color, boolean enabled)
```

Comment:
Unless I missed something looking at the diff, this method probably can be static.